cli/tests/extensions.rs

```diff
@@ -280,6 +280,135 @@ fn conflicting_extension_name_is_filtered() {
     assert!(output.contains("extension was filtered out"));
 }
 
+mod permissions {
+    use std::ffi::OsStr;
+
+    use assert_cmd::assert::Assert;
+    use predicates::prelude::*;
+
+    use super::*;
+
+    struct TestCLI {
+        tempdir: TempDir,
+        cwd: Option<PathBuf>,
+    }
+
+    impl Default for TestCLI {
+        fn default() -> Self {
+            Self { tempdir: TempDir::new().unwrap(), cwd: None }
+        }
+    }
+
+    impl TestCLI {
+        fn new() -> Self {
+            Default::default()
+        }
+
+        fn cwd(mut self, cwd: PathBuf) -> Self {
```

Comment:
```suggestion
        fn cwd(&mut self, cwd: PathBuf) -> &mut Self {
```

Pretty sure this is the more common way to do this in Rust, this way the API can be used either way.